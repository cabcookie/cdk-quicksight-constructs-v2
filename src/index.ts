import { RemovalPolicy } from 'aws-cdk-lib';
import { Bucket } from 'aws-cdk-lib/aws-s3';
import { Construct } from 'constructs';

// export interface QuickSightTestConstructProps {

// }

export class QuickSightTestConstruct extends Construct {
  public readonly bucketName: string;

  constructor(scope: Construct, id: string) {
    super(scope, id);

    const bucket = new Bucket(this, 'QSCTestBucket', {
      removalPolicy: RemovalPolicy.DESTROY,
    });

    this.bucketName = bucket.bucketName;
  }
}