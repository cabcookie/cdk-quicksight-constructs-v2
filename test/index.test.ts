import { App, Stack } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { QuickSightTestConstruct } from '../src';

test('Construct Snapshot', () => {
  const app = new App();
  const stack = new Stack(app, 'MyTestStack');
  new QuickSightTestConstruct(stack, 'MyQSTest');

  const assert = Template.fromStack(stack);
  const resource = assert.findResources('AWS::S3::Bucket');

  expect(resource).toMatchSnapshot();
});