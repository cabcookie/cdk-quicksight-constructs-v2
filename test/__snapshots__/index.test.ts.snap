// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Construct Snapshot 1`] = `
Object {
  "MyQSTestQSCTestBucket118BB3F7": Object {
    "DeletionPolicy": "Delete",
    "Type": "AWS::S3::Bucket",
    "UpdateReplacePolicy": "Delete",
  },
}
`;
